# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [1]:
library(tidyverse) 

-- Attaching packages --------------------------------------- tidyverse 1.2.1 --
<U+221A> ggplot2 2.2.1     <U+221A> purrr   0.2.4
<U+221A> tibble  1.4.2     <U+221A> dplyr   0.7.4
<U+221A> tidyr   0.8.0     <U+221A> stringr 1.3.0
<U+221A> readr   1.1.1     <U+221A> forcats 0.3.0
-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()


# Assignment 1

First, we generate some random data:

In [4]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5        V6        V7       
1 0.5354232 0.2249460 0.6477022 0.6883457 0.3812886 0.2758890 0.2876209
2 0.1800904 0.8042258 0.9994616 0.7289843 0.4726363 0.1758506 0.7292607
  V8         V9        V10       V11       V12        V13       V14      
1 0.09693642 0.9458573 0.6466998 0.2040720 0.01683959 0.8747655 0.1255905
2 0.85036690 0.7817908 0.6698567 0.4420549 0.67375614 0.6784593 0.4339054
  V15       V16       V17       V18       V19       V20        
1 0.6932493 0.2429624 0.7138738 0.3486898 0.7524257 0.566530329
2 0.9833230 0.7546722 0.2838772 0.4781837 0.5514137 0.008567376

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [7]:
df2 <- df1 %>% gather(Key, Value, V1:V20)
  

print(df2)

   Key       Value
1   V1 0.535423228
2   V1 0.180090417
3   V2 0.224946007
4   V2 0.804225798
5   V3 0.647702222
6   V3 0.999461638
7   V4 0.688345714
8   V4 0.728984343
9   V5 0.381288595
10  V5 0.472636258
11  V6 0.275888973
12  V6 0.175850569
13  V7 0.287620883
14  V7 0.729260710
15  V8 0.096936421
16  V8 0.850366902
17  V9 0.945857330
18  V9 0.781790836
19 V10 0.646699839
20 V10 0.669856742
21 V11 0.204072036
22 V11 0.442054905
23 V12 0.016839594
24 V12 0.673756143
25 V13 0.874765506
26 V13 0.678459318
27 V14 0.125590453
28 V14 0.433905418
29 V15 0.693249263
30 V15 0.983322959
31 V16 0.242962388
32 V16 0.754672164
33 V17 0.713873809
34 V17 0.283877190
35 V18 0.348689833
36 V18 0.478183725
37 V19 0.752425736
38 V19 0.551413742
39 V20 0.566530329
40 V20 0.008567376


## Assignment 2

We will generate some data:

In [8]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [5]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [6]:
df5 <- full_merge(df3, df4, by)
df5

ERROR: Error in eval(expr, envir, enclos): object 'df5' not found


End of notebook